File: Unity/Assets/Scripts/Classes/Deck/Card/Story/storyCard.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public abstract class storyCard : Card{

	
	
    protected string storyCardType;	
    //Constructor
	public storyCard(string name, string storyCardType) : base(name, "STORYCARD")
    {
	
	this.storyCardType = storyCardType;
	
	}

   public string getStoryType(){return storyCardType;}	
	
}

File: Java/Maven/src/main/java/com/COMP3004/Maven/Story/storyCard.java
package com.COMP3004.Maven.Story;

import java.io.Serializable;
import java.util.List;

import com.COMP3004.Maven.Card;
import com.COMP3004.Maven.Player;
import com.COMP3004.Maven.TableDeck.AdventureDeck;

public abstract class storyCard extends Card implements Serializable {

	
	
	private static final long serialVersionUID = 1L;
	
	storyBehaviour behaviour;

	// Constructor
	public storyCard(String type, String name) {
		super("STORY", type, name);
	}

	public void play(Player current, List<Player> players, AdventureDeck aD, int bonusShields) {
		behaviour.play(current, players, aD, bonusShields);
	}
}
